src/ZendeskApi.Client/Resources/HttpRequestExtensions.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
```

Comment:
I have to say that I do agree with `HttpChannel` being a bit of a mess. I didn't originally work on this client and I can't quite see the use case either. Perhaps we just sweep this under the rug for now 😉